src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1194,29 +1194,27 @@ class TreeDataSource implements IAsyncDataSource<ITreeItem, ITreeItem> {
 		return !!this.treeView.dataProvider && (element.collapsibleState !== TreeItemCollapsibleState.None);
 	}
 
-	private batch: ITreeItem[] | undefined;
-	private batchPromise: Promise<ITreeItem[][] | undefined> | undefined;
+	private batch: { elements: ITreeItem[]; promise: Promise<ITreeItem[][] | undefined> | undefined } | undefined;
 	async getChildren(element: ITreeItem): Promise<ITreeItem[]> {
 		const dataProvider = this.treeView.dataProvider;
 		if (!dataProvider) {
 			return [];
 		}
 		if (this.batch === undefined) {
-			this.batch = [element];
-			this.batchPromise = undefined;
+			this.batch = { elements: [element], promise: undefined };
 		} else {
-			this.batch.push(element);
+			this.batch.elements.push(element);
 		}
-		const indexInBatch = this.batch.length - 1;
+		const indexInBatch = this.batch.elements.length - 1;
+		const batch = this.batch;
```

Comment:
I may be wrong, but I think for the batching to be effective we must capture the batch within the `settimeout`.